### PR TITLE
Remove useless PG version compats

### PR DESCRIPTION
**Pull request by onderkalaci**
Lets get this before PG 15 changes to simplify compats

[Remove useless version compats](https://github.com/citusdata/citus/commit/7df6130e2bae4706a6f3c984156443fde9fea96c): While dropping old functions, we never cleaned up old compats

Also [Remove unusued gitattributes](https://github.com/citusdata/citus/commit/42c6d03dc6bca286ac8bb339368ea4c1341939e7)

Addresses https://github.com/citusdata/citus/pull/6040#discussion_r925598863

### CODE REVIEW

File: src/include/pg_version_compat.h [hunk 1]
```diff
@@ -123,37 +123,12 @@ RelationGetSmgr(Relation rel)
 #define ROLE_PG_READ_ALL_STATS DEFAULT_ROLE_READ_ALL_STATS
 #endif
 
-#define lnext_compat(l, r) lnext(l, r)
-#define list_delete_cell_compat(l, c, p) list_delete_cell(l, c)
-#define pg_plan_query_compat(p, q, c, b) pg_plan_query(p, q, c, b)
-#define planner_compat(p, c, b) planner(p, NULL, c, b)
-#define standard_planner_compat(a, c, d) standard_planner(a, NULL, c, d)
-#define GetSequencesOwnedByRelation(a) getOwnedSequences(a)
-#define GetSequencesOwnedByColumn(a, b) getOwnedSequences_internal(a, b, 0)
```
Comment (naisila) [inline src/include/pg_version_compat.h:132]: Maybe we can leave `GetSequencesOwnedByRelation` and `GetSequencesOwnedByColumn` definitions for readability. `getOwnedSequences_internal` might be confusing. Although, currently it has only one use in the codebase, and the variable is defined such that it avoids confusion:
```SQL
List *columnOwnedSequences = getOwnedSequences_internal(relationId, attributeIndex + 1);
```
What do you think?

**Review by naisila — COMMENTED** [review]
Does this line in `pg_version_compat.h` have a purpose? 
`#define ProcessCompletedNotifies()`

File: src/include/pg_version_compat.h (hunk 1 above)
Comment (onderkalaci) [reply src/include/pg_version_compat.h:132]: I think we should merge this PR, and then in another PR we could introduce better function names -- in another header.

In the end, `getOwnedSequences_internal` is Postgres' function, and tbh I don't think we should use another name for it?

Comment (naisila) [reply src/include/pg_version_compat.h:132]: Makes sense. Definitions like the following could also move to that header

**Review by naisila — APPROVED** [review]

**Pull request merged**